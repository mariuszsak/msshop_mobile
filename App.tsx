import React, { Component } from "react";
import AppTabNavigator from "./src/components/navigator/AppTabNavigator";
import { AppContext, ContextDefaultValue } from "./src/context";
import { fetchData } from "./src/services/fetchData";

type AppState = {
  items: [];
};

interface Props {
  navigation: any
}

export default class App extends Component<Props, AppState> {
  constructor(props: Props) {
    super(props);
    this.state = {
      items: []
    };
  }

  componentDidMount() {
    fetchData('products').then((response) => {
      this.setState({
        items: response
      });
    });

  }

  render() {
    return (
      <AppContext.Provider value={this.state}>
        <AppTabNavigator />
      </AppContext.Provider>
    );
  }
}
