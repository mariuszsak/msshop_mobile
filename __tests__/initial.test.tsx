import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import BasketComponent from "../src/components/basketComponent";
import Home from "../src/screens/home";


describe("Snapshot tests", () => {
  beforeEach(() => {
      jest.useFakeTimers();
    }
  );

  it("Snapshot test BasketComponent", () => {
    const tree = renderer.create(
      <BasketComponent name={"test"} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Snapshot test Home", () => {
    const tree = renderer.create(
      <Home navigation={null} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Snapshot test", () => {
    const tree = renderer.create(
      <App navigation={null} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
