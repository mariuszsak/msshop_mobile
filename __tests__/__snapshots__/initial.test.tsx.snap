// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "rgb(242, 242, 242)",
        },
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      collapsable={false}
      importantForAccessibility="auto"
      pointerEvents="auto"
      removeClippedSubviews={false}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNCSafeAreaProvider
            onInsetsChange={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                />
                <View
                  accessibilityElementsHidden={false}
                  closing={false}
                  gestureVelocityImpact={0.3}
                  importantForAccessibility="auto"
                  onClose={[Function]}
                  onGestureBegin={[Function]}
                  onGestureCanceled={[Function]}
                  onGestureEnd={[Function]}
                  onOpen={[Function]}
                  onTransition={[Function]}
                  pointerEvents="box-none"
                  style={
                    Array [
                      Object {
                        "overflow": undefined,
                      },
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                    ]
                  }
                  transitionSpec={
                    Object {
                      "close": Object {
                        "animation": "spring",
                        "config": Object {
                          "damping": 500,
                          "mass": 3,
                          "overshootClamping": true,
                          "restDisplacementThreshold": 10,
                          "restSpeedThreshold": 10,
                          "stiffness": 1000,
                        },
                      },
                      "open": Object {
                        "animation": "spring",
                        "config": Object {
                          "damping": 500,
                          "mass": 3,
                          "overshootClamping": true,
                          "restDisplacementThreshold": 10,
                          "restSpeedThreshold": 10,
                          "stiffness": 1000,
                        },
                      },
                    }
                  }
                >
                  <View
                    needsOffscreenAlphaCompositing={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "flex": 1,
                        "marginTop": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      forwardedRef={[Function]}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                            Object {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 1,
                              "width": -1,
                            },
                            "shadowOpacity": 0.3,
                            "shadowRadius": 5,
                            "top": 0,
                            "width": 3,
                          }
                        }
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            Array [
                              Object {
                                "backgroundColor": "rgb(242, 242, 242)",
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column-reverse",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <Text>
                                Loading
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "height": 64,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1,
                }
              }
            />
          </RNCSafeAreaProvider>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "bottom": 0,
        "elevation": 4,
        "left": 0,
        "position": null,
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(33, 150, 243, 1)",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "width": "100%",
            },
            Object {
              "marginBottom": 0,
              "maxWidth": 480,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": "#2196f3",
              "borderRadius": 0,
              "height": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "top": 28,
              "transform": Array [
                Object {
                  "scale": 0.008,
                },
              ],
              "width": 0,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#fff",
                        },
                      ],
                    ]
                  }
                >
                  Home
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 0.5,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#fff",
                        },
                      ],
                    ]
                  }
                >
                  Basket
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 0.5,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#fff",
                        },
                      ],
                    ]
                  }
                >
                  Filter
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 0.5,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#fff",
                        },
                      ],
                    ]
                  }
                >
                  My Account
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 0.5,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "#fff",
                        },
                      ],
                    ]
                  }
                >
                  Info
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Snapshot tests Snapshot test BasketComponent 1`] = `
<View>
  <Text>
    test
  </Text>
</View>
`;

exports[`Snapshot tests Snapshot test Home 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text>
    Loading
  </Text>
</View>
`;
